src/sock.c

```diff
@@ -99,6 +100,17 @@ sock_t sock_connect(const char * const host, const unsigned int port)
         if (sock < 0)
             continue;
 
+	/* turn on keepalive */
```

Comment:
The issue is mostly with 3G/4G operators using NAT. To keep things running, they set NAT connection tracking idle timeout to some small values, like 5 minutes. Connection beetween client and server seems to be active for both, but connection tracking record in operator's NAT already dropped. Simpiest way to avoid this is to use TCP keepalive. with short period.

However, I have no idea how to correctly add this feature to libstrophe. Shall it be some setting for connection, or additional function for socket - it's up to authors. I just tried to show correct place and method for it.